main.go

```diff
@@ -196,16 +313,31 @@ func main() {
 		}
 	}
 
-	subj := "faas-request"
-	qgroup = "faas"
-
-	log.Println("Wait for ", config.AckWait)
-	sub, err := sc.QueueSubscribe(subj, qgroup, mcb, startOpt, stan.DurableName(durable), stan.MaxInflight(config.MaxInflight), stan.AckWait(config.AckWait))
-	if err != nil {
-		log.Panicln(err)
+	ctx, cancelCtx := context.WithCancel(context.Background())
+
+	natsQueue := NatsQueue{
+		clusterID: "faas-cluster",
+		clientID:  "faas-worker-" + nats.GetClientID(hostname),
+		natsURL:   "nats://" + config.NatsAddress + ":4222",
+
+		connMutex:      &sync.RWMutex{},
+		maxReconnect:   config.MaxReconnect,
+		reconnectDelay: config.ReconnectDelay,
+		context:        ctx,
+		cancelContext:  cancelCtx,
```

Comment:
I guess that this may be the right way to do it with context, but IMO, I find this more complex than just having a simple `quitCh: make(chan struct{})` and on connection close, simply close that channel. You would still call `case <-q.quitCh:` in `reconnect()`, which when closed will trigger the `case`. You don't need to check if it is closed or anything since the only thing you do with it is close it.
But again, this is just coding preferences, but I personally do not use the context so not entirely sure looking at the code if that would work. Hope you have a test for that :-)

Reply:
Make sense. Let me do that this way

Reply:
Done! :-)